core/src/View/Kokkos_ViewLegacy.hpp

```diff
@@ -1506,9 +1511,10 @@ struct CommonViewAllocProp;
 
 template <class ValueType>
 struct CommonViewAllocProp<void, ValueType> {
-  using value_type        = ValueType;
+  using value_type = ValueType;
+#ifdef KOKKOS_ENABLE_DEPRECATED_CODE_4
   using scalar_array_type = ValueType;
```

Comment:
Is it observable?  Used anywhere?

Reply:
Looking at Trilinos, the Sacado package uses that type. Changes to handle that on their side would be trivial.

Reply:
I meant is that public, is there a legal way for users to refer to that type.

Reply:
Yes. In practice nobody probably does. The automatic differentiation thingy in Trilinos provides a specialized `CommonViewAllocProp` where they then access the `value_type`. The deprecated code also seems to be wrong already as it assigned ValueType to scalar_array_type.